pkg/roachprod/testselector/roachtestselector/consumer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// consumer is responsible for reading the csv files containing the test details generated by testselector
+
+package roachtestselector
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/errors"
+	"google.golang.org/api/option"
+)
+
+// ReadTestsToRun reads the tests to run based on certain criteria:
+// 1. the number of time a test has been successfully running
+// 2. the test is new
+// 3. the test has not been run for a while
+// 4. a subset of the successful tests
+// It returns a new list of selected tests. The individual tests in the input
+// are also modified with the skip reason.
+// It also returns the number of tests that satisfied the selection criteria.
+func ReadTestsToRun(
+	ctx context.Context, tests []registry.TestSpec, cloud, suite string,
+) ([]registry.TestSpec, int, error) {
+	options := []option.ClientOption{option.WithScopes(storage.ScopeReadOnly), option.WithQuotaProject(project)}
+	cj := os.Getenv("GOOGLE_EPHEMERAL_CREDENTIALS")
+	if len(cj) != 0 {
+		options = append(options, option.WithCredentialsJSON([]byte(cj)))
+	} else {
+		fmt.Printf("GOOGLE_EPHEMERAL_CREDENTIALS env is not set.\n")
```

Comment:
Maybe exit early or return an error? 

It doesn't make much sense to still try to read the thing if we don't have access to the credentials. Unless you're going to make the thing publicly accessible, but then we shouldn't be trying to use the credentials anyway.

Reply:
If I am running this locally, it works without the environment variable being set as I am already loggedin in my console. So, I have this.